editor/plugins/theme_editor_plugin.cpp

```diff
@@ -2370,18 +2370,25 @@ void ThemeTypeEditor::_update_type_list_debounced() {
 	update_debounce_timer->start();
 }
 
-HashMap<StringName, bool> ThemeTypeEditor::_get_type_items(String p_type_name, void (Theme::*get_list_func)(StringName, List<StringName> *) const, bool include_default) {
+HashMap<StringName, bool> ThemeTypeEditor::_get_type_items(String p_type_name, Theme::DataType p_type, bool p_include_default) {
 	HashMap<StringName, bool> items;
 	List<StringName> names;
 
-	if (include_default) {
+	if (p_include_default) {
 		names.clear();
 		String default_type = p_type_name;
 		if (edited_theme->get_type_variation_base(p_type_name) != StringName()) {
 			default_type = edited_theme->get_type_variation_base(p_type_name);
 		}
 
-		(ThemeDB::get_singleton()->get_default_theme().operator->()->*get_list_func)(default_type, &names);
+		List<ThemeDB::ThemeItemBind> theme_binds;
+		ThemeDB::get_singleton()->get_class_own_items(default_type, &theme_binds);
+		for (const ThemeDB::ThemeItemBind &E : theme_binds) {
+			if (E.data_type == p_type) {
+				names.push_back(E.item_name);
+			}
+		}
```

Comment:
It's probably worth it to add a new method to filter by the data type and avoid excessive loops.

Reply:
Looks like it's needed in https://github.com/godotengine/godot/pull/85015 as well.